lib/travis/build/script/r.rb

```diff
@@ -115,6 +124,9 @@ def announce
 
         def install
           super
+          unless setup_cache_has_run_for[:packages]
```

Comment:
This seems a bit unconventional for Ruby code. There is no need for a hash here. For example, an instance variable (e.g., `@cache_setup_has_run`) should do.